### PR TITLE
Split mono-repo in different `npm` packages

### PR DESCRIPTION
**Pull request by anders-kiaer**
Partially solves #1493.

- [X] Move component source code to separate folders. @anders-kiaer 
- [X] Split runtime dependencies in current single `package.json` into each components `package.json`. @anders-kiaer 
- [X] Use ~~`lerna`~~ `nx` @anders-kiaer 
- [X] One single `tsconfig.json` in root folder (looks like this is not fully possible https://github.com/microsoft/TypeScript/issues/25430, but we can put most of the config in the root folder by extending it in each package). @anders-kiaer 
- [x] Keep dev-dependencies in root for consistency across packages (?). @anders-kiaer 
- [x] Ensure build of common components runs to completion... i.e. `npx nx build group-tree` or `npx nx run group-tree:build` (and subsurface-viewer, well-completions, well-log-viewer) @hkfb 
- [x] Ensure generated packages are usable... still need to find how to generate publishable libraries :/
- [X] Rename top-folder from `/react` to `/typescript`.

After discussion with @hkfb we think that we will postpone the following three tasks to a later PR:
- Adjust storybook setup such that we can keep ONE storybook to publish containing stories of all the components.
- Adjust CI setup to succeed with new structure.
- Adjust Dash build setup to work with `nx` and multiple packages.
- Remove unused (dev)dependencies in root workspace `package.json`

### CODE REVIEW

**Comment by t0oF-azpn** [comment]
there is still a react/src/lib directories holding files. I guess these need to be handled and dispatched to the corresponding workspaces.

File: react/packages/subsurface-viewer/package.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{
+    "name": "subsurface-viewer",
```
[the remark this answers is not in the document]
Comment (anders-kiaer) [reply react/packages/subsurface-viewer/package.json]: Changed

**Pull request merged**